src/metpy/plots/patheffects.py

```diff
@@ -0,0 +1,250 @@
+        segment_indices, marker_offsets = self._get_marker_locations(offsets, renderer)
+
+        # Draw the original path
+        renderer.draw_path(gc0, path, affine, rgbFace)
+
+        # Need to account for the line width in order to properly draw symbols at line edge
+        line_shift = renderer.points_to_pixels(gc.get_linewidth()) / 2
+
+        # Loop over all the markers to draw
+        for ind, marker_offset in zip(segment_indices, marker_offsets):
+            sym_trans = self._get_symbol_transform(renderer, marker_offset, line_shift,
+                                                   angles[ind], starts[ind])
+            renderer.draw_path(gc0, self._symbol, sym_trans, self.color)
+
+        gc0.restore()
+
+
+@exporter.export
+class ColdFront(Front):
+    """Draw a path as a cold front, with (default blue) pips/triangles along the path."""
+
+    _symbol = mpath.Path([[0, 0], [1, 1], [2, 0], [0, 0]],
+                         [mpath.Path.MOVETO, mpath.Path.LINETO, mpath.Path.LINETO,
+                          mpath.Path.CLOSEPOLY])
+
+    def __init__(self, color='blue', **kwargs):
+        super().__init__(color, **kwargs)
+
+
+@exporter.export
+class WarmFront(Front):
+    """Draw a path as a warm front with (default red) scallops along the path."""
+
+    _symbol = mpath.Path.wedge(0, 180).transformed(mtransforms.Affine2D().translate(1, 0))
+
+    def __init__(self, color='red', **kwargs):
+        super().__init__(color, **kwargs)
+
+
+@exporter.export
+class OccludedFront(Front):
+    """Draw an occluded front with (default purple) pips and scallops along the path."""
+
+    def __init__(self, color='purple', **kwargs):
+        self._symbol_cycle = None
+        super().__init__(color, **kwargs)
+
+    def draw_path(self, renderer, gc, path, affine, rgbFace=None):  # noqa: N803
+        """Draw the given path."""
+        self._symbol_cycle = None
+        return super().draw_path(renderer, gc, path, affine, rgbFace)
+
+    @property
+    def _symbol(self):
+        """Return the proper symbol to draw; alternatives between scallop and pip/triangle."""
+        if self._symbol_cycle is None:
+            self._symbol_cycle = itertools.cycle([WarmFront._symbol, ColdFront._symbol])
+        return next(self._symbol_cycle)
+
+
+@exporter.export
+class StationaryFront(Front):
+    """Draw a stationary front as alternating cold and warm front segments."""
+
+    _symbol = WarmFront._symbol
+    _symbol2 = ColdFront._symbol.transformed(mtransforms.Affine2D().scale(1, -1))
+
+    def __init__(self, colors=None, **kwargs):
+        """Initialize a stationary front path effect.
+
+        Parameters
+        ----------
+        colors : Sequence[str] or Sequence[tuple[float]]
+            Matplotlib color identifiers to cycle between on the two different front styles
+        """
+        if colors is None:
+            colors = ['red', 'blue']
+        self._colors = list(map(mcolors.to_rgba, colors))
```

Comment:
Not to drag it out, but some additional thoughts when I look at the rendered docs:
* This fix wipes out the inherited init docstring, so loses docs for the other kwargs
* Color default values aren't documented here, and it's not clear which color maps to which symbol. Can we define the default colors in the signature and skip 196:197?
* Led me to realize we have potentially confusing (though not strictly wrong heh) documented defaults in the other subclasses:
<img width="512" alt="image" src="https://user-images.githubusercontent.com/24628426/230669402-330c7195-d548-4c20-8dff-b304412140b6.png">

Reply:
Ok, I think I covered that:
* Removed default of black for `Front`
* Just put the default stationary colors as a tuple in `__init__` signature
* Copied the rest of the docstring into `StationaryFront`
* Clarified the colors a bit.